dbms/src/Functions/FunctionsVisitParam.h

```diff
@@ -87,61 +87,45 @@ struct ExtractBool
 
 struct ExtractRaw
 {
-    static void extract(const UInt8 * pos, const UInt8 * end, ColumnString::Chars_t & res_data)
+    inline static void skipAfterQuotationIfNeed(const UInt8 *& pos, const UInt8 * end, ColumnString::Chars_t & res_data)
     {
-        if (pos == end)
-            return;
-
-        UInt8 open_char = *pos;
-        UInt8 close_char = 0;
-        switch (open_char)
+        if (pos + 1 < end && pos[1] == '"')
         {
-            case '[':
-                close_char = ']';
-                break;
-            case '{':
-                close_char = '}';
-                break;
-            case '"':
-                close_char = '"';
-                break;
+            res_data.push_back(*pos);
+            ++pos;
         }
+    }
 
-        if (close_char != 0)
-        {
-            size_t balance = 1;
-            char last_char = 0;
-
-            res_data.push_back(*pos);
+    static void extract(const UInt8 * pos, const UInt8 * end, ColumnString::Chars_t & res_data)
+    {
+        std::vector<char> expect_end;
```

Comment:
It will lead to allocation/deallocation inside a loop.
We can use PODArray with AllocatorWithStackMemory with for example, 64 bytes of automatic memory.